swig/cpp/src/Tree_Data.cpp

```diff
@@ -520,35 +520,50 @@ S_Value Attr::value() {
 }
 S_Attr Attr::next() LY_NEW(attr, next, Attr);
 
-Difflist::Difflist(struct lyd_difflist *diff, S_Deleter deleter) {
-    diff = diff;
-    deleter = std::make_shared<Deleter>(diff, deleter);
+Difflist::Difflist(struct lyd_difflist *diff, S_Deleter deleter)
+  : diff(diff)
+  , deleter(std::make_shared<Deleter>(diff, deleter))
+{
 }
 Difflist::~Difflist() {};
 std::vector<S_Data_Node> Difflist::first() {
     std::vector<S_Data_Node> s_vector;
     unsigned int i = 0;
+    unsigned int s = 0;
+
+    for(s = 0; diff->type[s] != LYD_DIFF_END; s++) {
+    }
+
+    s_vector.reserve(s);
 
-    if (!*diff->first) {
+    if (diff->first == nullptr) {
```

Comment:
```c++
if (!diff->first) {
```

Reply:
I like this :-)

Reply:
Yes, it is the same, and therefore it should be made similar to what the rest of the code uses.